src/runtime/RopeString.cpp

```diff
@@ -187,6 +187,27 @@ class RopeStringUsageChecker {
     bool m_headOfCharAt;
 };
 
+// worker for preventing too many recursive calls with RopeString
+static char16_t charAtWorker(size_t idx, const RopeString* self)
+{
+ReadStart:
+    size_t leftLength = self->left()->length();
+    if (idx < leftLength) {
+        if (self->left()->isRopeString() && !self->left()->asRopeString()->wasFlattened()) {
+            self = self->left()->asRopeString();
+            goto ReadStart;
```

Comment:
IMHO I think that its better to avoid `goto` statement as much as possible if its not inevitable.
In this case, it seems that these `goto-label` statements could be replaced by a loop.
What do you think about this?

Reply:
I replaced goto with while statement